exercises/affine-cipher/description.md

```diff
@@ -10,61 +10,54 @@ its new numeric value. Although all mono-alphabetic ciphers are weak,
 the affine cipher is much stronger than the atbash cipher,
 because it has many more keys.
 
+## Encryption
+
 The encryption function is:
 
-  `E(x) = (ax + b) mod m`
-  -  where `x` is the letter's index from 0 - length of alphabet - 1
-  -  `m` is the length of the alphabet. For the roman alphabet `m == 26`.
-  -  and `a` and `b` make the key
+  `E(x) = (ai + b) mod m`
+  -  where `i` is the letter's index from `0` to the length of the alphabet - 1
+  -  `m` is the length of the alphabet. For the Roman alphabet `m` is `26`.
+  -  `a` and `b` are integers which make the encryption key
+
+Values `a` and `m` must be *coprime* (or, *relatively prime*) for automatic decryption to succeed,
+ie. they have number `1` as their only common factor (more information can be found in the
+[Wikipedia article about coprime integers](https://en.wikipedia.org/wiki/Coprime_integers)). In case `a` is
+not coprime to `m`, your program should return `"Error: a and m must be coprime."`. Otherwise it should
```

Comment:
I'm not the happiest with the word "return" here. In some languages, it is incorrect use of the type system to return a string value to signify an error. Instead, they might raise an exception, or return a None variant of their Option, or return an Left variant of their Either, or return an Err variant of their Error, or return `{:error, reason}`. For this reason, I advise against using the word "return" here. The suggestion I make would allow languages to do what is best for them.

```suggestion
not coprime to `m`, your program should indicate that this is an error. Otherwise it should
```